### PR TITLE
refactor: move example rasters to test fixtures and default data directory to `workers`

### LINKED ISSUE #311: Move example raster on which the tests are depending to fixture directory

### PR DESCRIPTION
**Pull request by matthiasschaub**
### Description
Move example rasters to test fixtures.
Remove data directory.
Change default data dir to be in workers.

### Corresponding issue
Closes #311 

### New or changed dependencies
- None

### Checklist
- [x] I have updated my branch to `main` (e.g. through `git rebase main`)
- [x] My code follows the [style guide](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#style-guide) and was checked with [pre-commit](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#pre-commit) before committing
~- [ ] I have commented my code~
- [x] I have added sufficient unit and integration [tests](https://github.com/GIScience/ohsome-quality-analyst/blob/main/docs/development_setup.md#tests)
- [x] I have updated the [CHANGELOG.md](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CHANGELOG.md)

### CODE REVIEW

**Review by joker234 — CHANGES_REQUESTED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -79,7 +79,7 @@ pipeline {
                                 -e POSTGRES_DB=oqt \
                                 -e POSTGRES_USER=oqt \
                                 -e POSTGRES_PASSWORD=oqt""") { c ->
-              WORKERS_CI.inside("--network ${n} -e POSTGRES_HOST=${POSTGRES_HOST} -e POSTGRES_PORT=${POSTGRES_PORT} -e OQT_DATA_DIR=/data -v ${WORKSPACE}/data:/data") {
+              WORKERS_CI.inside("--network ${n} -e POSTGRES_HOST=${POSTGRES_HOST} -e POSTGRES_PORT=${POSTGRES_PORT}") {
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply Jenkinsfile:82]: All tests are mocking the reading of the raster files found in the data directory. So no need of this directory if only tests are running. 
Should their be one integration test which reads from this directory?

File: docs/raster_datasets.md [hunk 2]
```diff
@@ -1,23 +1,20 @@
...
+OQT operates on raster datasets stored on disk. Those raster datasets need to be
+downloaded and saved to the OQT data directory.
 
 ## Location (Data directory)
 
-The raster files are located in the `data` directory at the root of git repository.
-
-The location can be changed by setting the environment variable `$OQT_DATA_DIR`.
+The default data directory is `workers/data`. The location can be changed by setting the
+environment variable `$OQT_DATA_DIR`.
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply docs/raster_datasets.md:9]: Done

File: workers/ohsome_quality_analyst/config.py [hunk 3]
```diff
@@ -106,13 +106,14 @@ def get_default_data_dir() -> str:
...
-        "..",
-        "data",
+    return os.path.abspath(
+        os.path.join(
+            os.path.dirname(
+                os.path.abspath(__file__),
+            ),
+            "..",
+            "data",
+        )
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply workers/ohsome_quality_analyst/config.py]: Done: See `helper.get_project_root()`

File: workers/tests/integrationtests/test_indicator_building_completeness.py [hunk 4]
```diff
@@ -15,97 +14,94 @@
...
-        with self.assertRaises(HexCellsNotFoundError):
+from .utils import get_fixture_dir, get_geojson_fixture, get_layer_fixture, oqt_vcr
+
+
+@pytest.fixture
+def mock_env_oqt_data_dir(monkeypatch):
+    directory = os.path.join(get_fixture_dir(), "rasters")
+    monkeypatch.setenv("OQT_DATA_DIR", directory)
+
+
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply workers/tests/integrationtests/test_indicator_building_completeness.py:25]: I would like to get away from unittest and use pytest. I would only use classes if they are useful in someway or another.

File: workers/tests/integrationtests/test_indicator_building_completeness.py [hunk 5]
```diff
@@ -15,97 +14,94 @@
...
-        result = asyncio.run(get_shdi(FeatureCollection(features=[self.feature])))
-        self.assertIsInstance(result, list)
-        self.assertEqual(len(result), 1)
+    def test_get_shdi(feature):
+        result = asyncio.run(get_shdi(FeatureCollection(features=[feature])))
+        assert isinstance(result, list)
+        assert len(result) == 1
 
 
 if __name__ == "__main__":
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply workers/tests/integrationtests/test_indicator_building_completeness.py:107]: Removed and also removed unused import statement

File: workers/tests/integrationtests/test_indicator_ghs_pop_comparison_roads.py [hunk 6]
```diff
@@ -1,46 +1,48 @@
...
-
-if __name__ == "__main__":
-    unittest.main()
+from .utils import get_fixture_dir, get_geojson_fixture, get_layer_fixture, oqt_vcr
+
+
+@pytest.fixture
+def mock_env_oqt_data_dir(monkeypatch):
+    directory = os.path.join(get_fixture_dir(), "rasters")
+    monkeypatch.setenv("OQT_DATA_DIR", directory)
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply workers/tests/integrationtests/test_indicator_ghs_pop_comparison_roads.py:17]: Factor out this to utills.py

File: workers/tests/integrationtests/test_indicator_ghs_pop_comparison_buildings.py [hunk 7]
```diff
@@ -1,44 +1,48 @@
...
+def mock_env_oqt_data_dir(monkeypatch):
+    directory = os.path.join(get_fixture_dir(), "rasters")
+    monkeypatch.setenv("OQT_DATA_DIR", directory)
+
+
+@oqt_vcr.use_cassette()
+def test(mock_env_oqt_data_dir):
+    feature = get_geojson_fixture("algeria-touggourt-feature.geojson")
+    layer = get_layer_fixture("building_count")
+    indicator = GhsPopComparisonBuildings(feature=feature, layer=layer)
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply workers/tests/integrationtests/test_indicator_ghs_pop_comparison_buildings.py]: No reason other than it was implemented as unittest before the decision to use pytest

File: CHANGELOG.md [hunk 8]
```diff
@@ -31,6 +32,7 @@
 - Substitute Simple Report with a Report named Minimal for testing purposes ([#342] [#385])
 - Add a minimal Indicator for testing purposes ([#383])
 - Remove database scripts ([#392])
+- Move example rasters of data directory to test fixtures ([#312])
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply CHANGELOG.md]: I did not add it to How To Updgrade section because I thought this section is for end-users not users who setup OQT on their own.

**Review by joker234 — APPROVED** [review]

**Review by joker234 — APPROVED** [review]

**Pull request merged**